e2e-test/clients.py

```diff
@@ -38,6 +40,26 @@ def read_file(file_name: str) -> Dict[str, Any]:
 
     return data
 
+def parse_date(date_str):
```

Comment:
So as discussed off-PR, please rewrite this function with a custom date-parsing function and avoid issues of `datetime.fromisoformat`.